templates/go/cloudevents/handle.go

```diff
@@ -20,24 +20,28 @@ func Handle(ctx context.Context, event cloudevents.Event) (*event.Event, error)
 	 */
 
 	fmt.Printf("Incoming Event: %v\n", event) // print the received event to standard output
-	payload := ""
+	var payload Echo
 	err := json.Unmarshal(event.Data(), &payload)
 	if err != nil {
 		fmt.Printf("%v\n", err)
 		return nil, err
 	}
 
-	payload = "echo " + payload
+	payload.Message = "echo " + payload.Message
 	outputEvent := cloudevents.NewEvent()
 	outputEvent.SetSource("http://example.com/echo")
 	outputEvent.SetType("Echo")
-	outputEvent.SetData(cloudevents.ApplicationJSON, &payload)
+	outputEvent.SetData(cloudevents.ApplicationJSON, payload)
 
 	fmt.Printf("Outgoing Event: %v\n", outputEvent)
 
 	return &outputEvent, nil
 }
 
+type Echo struct {
+	Message string `json:"message"`
+}
+
```

Comment:
I am gong to approve this for now, but my general opinion is that the default "echo" handlers should be two lines:
1) print the incoming event to log
2) return the received event.  That is an Echo.

I am grateful for the improvement, but I feel compelled to ask the question: why is this handler, which should be the epitome of simplicity, continueing to grow?  This is perhaps a topic for a future discussion, but an "Echo" is literally "reflect, as in a mirror, what was received".  Why can this handler not simply be:
```
func Handle(ctx context.Context, event cloudevents.Event) (*event.Event, error) {
  fmt.Println(event)
  return &event, nil
}
```

Reply:
Thanks for your feedback @lkingland. Personally I prefer a template as simple as possible, as it was on previous commits. Looking to the history and the direction the template took I realized  that apply a fix would be more appropriate (instead of rolling back), hence my initial goal in this PR was to get E2E tests covered without changing (at least change much on the templates). But I'd be more than welcome to makethe template simpler (my preference) and leave the sample implementation (with json parsing) in  a different place (let's say on 'func-tastic') as long as @salaboy is ok with that!